src/simulations/data_collector.jl

```diff
@@ -180,4 +180,213 @@ function series_replicates(model, agent_step!, model_step!, properties, when, n,
     end
   end
   return dataall
+end
+
+
+"""
+    paramscan(;parameters::Dict, properties, when::AbstractArray, initialize, agent_step, model_step)
+
+Runs the model with all the parameter value combinations given in `parameters`.
+`parameters` is a dictionary that maps parameter names (symbol) to parameter
+values. If you want to test a range of parameters, you should specify those ranges in a 
+`Vector`.
+
+`initialize` is a function that creates an ABM object. It should accepts keyword arguments.
+It should have arguments for every key in the `parameters` dict, even if it does not use them..
+
+`properties` is the same dictionary used in the `step!` function that determines
+what information should be collected.
+
+Running replicates is not implemented, yet.
+
+# Example
+
+Here is an example of using `paramscan` with the Schelling model:
```

Comment:
please put this in the `tutorial.md` file as runnable Julia code. Add one more `@example` block.

This is just way too big for a documentation string. Let's reuse the fact that Tutorial.md already defines the schelling model...